demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -1647,10 +1937,27 @@ int main( int argc,
                 transportInterface.pNetworkContext = &networkContext;
             }
 
-            /******************** Download S3 Object File. **********************/
+            /******************** Upload S3 Object File. **********************/
 
             pPath = AWS_S3_URI_PATH;
 
+            if( returnStatus == EXIT_SUCCESS )
+            {
+                ret = uploadS3ObjectFile( &transportInterface,
+                                            pPath );
+                returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            }
+
+            /******************* Verify S3 Object File Upload. ********************/
+            if( returnStatus == EXIT_SUCCESS )
+            {
+                /* Verify the file exists by retrieving the file size. */
+                ret = verifyS3ObjectFileSize( &transportInterface,
+                                              pPath );
+                returnStatus = ( ret == true ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            }
+
+			/******************** Download S3 Object File. **********************/
             if( returnStatus == EXIT_SUCCESS )
             {
                 ret = downloadS3ObjectFile( &transportInterface,
```

Comment:
download the file to the local. we can do md5sum check and see the file is same to the local one.